mlir/include/mlir/Dialect/XeGPU/uArch/uArch.h

```diff
@@ -0,0 +1,389 @@
+  // uArchHierarchy[2] = {XeSlice, 16}
+  // uArchHierarchy[3] = {XeStack, 4}
+  // uArchHierarchy[4] = {gpu, 2}
+  std::vector<uArchHierarchyComponent> uArch_hierarchy;
+  // Different kind of regiger file information (e.g., GRF, ARF, etc.)
+  std::map<std::string, RegisterFileInfo> register_file_info;
+  // Each level of cache is indexed lower to higher in the vector
+  // (e.g., L1 indexed at 0, L2 at 1 and so on) L1, L2, L3, etc.
+  std::vector<CacheInfo> cache_info;
+  std::map<std::string, std::shared_ptr<Instruction>> instructions;
+  std::vector<Restriction<> *> restrictions;
+
+  // Constructor
+  uArch() = default;
+  uArch(const std::string &name, const std::string &description,
+        const std::vector<uArchHierarchyComponent> &uArch_hierarchy = {},
+        const std::map<std::string, RegisterFileInfo> &register_file_info = {},
+        const std::vector<CacheInfo> &cache_info = {},
+        const std::map<std::string, std::shared_ptr<Instruction>>
+            &instructions = {},
+        const std::vector<Restriction<> *> &restrictions = {})
+      : name(name), description(description), uArch_hierarchy(uArch_hierarchy),
+        register_file_info(register_file_info), cache_info(cache_info),
+        instructions(instructions), restrictions(restrictions) {}
+};
+
+// A struct to represent shared memory information
+struct SharedMemory {
+  uint32_t size;      // in bytes
+  uint32_t alignment; // in bytes
+  // @TODO: Add more fields as needed
+  // uint32_t latency;
+  // uint32_t throughput;
+  // uint32_t bandwidth;
+  // uint32_t num_ports;
+  // uint32_t port_width;
+  // uint32_t bank_size;
+  // uint32_t bank_conflicts;
+  // uint32_t num_banks;
+
+  // Constructor
+  SharedMemory(uint32_t size, uint32_t alignment)
+      : size(size), alignment(alignment) {}
+};
+
+// For future use case in Xe4+
+
+// struct EUInfo {
+//     uint32_t num_eu_threads;
+//     SharedMemory shared_memory;
+// };
+
+//     uint32_t num_simd_units;
+//     uint32_t num_spus;
+//     uint32_t num_smt;
+//     uint32_t num_hardware_threads;
+//     uint32_t num_threads_per_spu;
+//     uint32_t num_threads_per_simd_unit;
+//     uint32_t num_threads_per_hardware_thread;
+//     uint32_t num_threads_per_smt;
+//     SharedMemory shared_memory;
+// };
+
+// A struct to represent a GPU uArch
+// This struct is used to represent the GPU microarchitecture of a target device
+// struct GPUuArch : public uArch {
+//     uint32_t num_compute_units;
+//     uint32_t num_vector_units;
+//     uint32_t num_scalar_units;
+//     uint32_t num_tensor_units;
+//     uint32_t num_matrix_units;
+//     SharedMemory shared_memory;
+// };
+
+// Create a TileLikeOp Interface
+struct TileOpInterface {
+  // Get the supported tiles for the specific data type.
+  // Can provide load/store/prefetch ops supported tile sizes for a specific
+  // uarch
+  virtual std::vector<std::vector<uint32_t>>
+  getSupportedTiles(mlir::Type type) = 0;
```

Comment:
getSupportedTiles => getSupportedShape

Reply:
Consider each interface has three basic functions: 
getSupportedShape
CheckSupportedType 
CheckSupportedShapes

Reply:
consider specific interface for :  chunked load, 1dload.